cucumber-expressions/ruby/lib/cucumber/cucumber_expressions/parameter_type_matcher.rb

```diff
@@ -47,12 +47,12 @@ def <=>(other)
 
       def space_before_match_or_sentence_start
         match_begin = @match.begin(0)
-        match_begin == 0 || @text[match_begin - 1].match(/\s/)
+        match_begin == 0 || @text[match_begin - 1].match(/\s|[[:punct:]]/)
```

Comment:
As far as I know `\s` only matches `[ \t\r\n\]`. However there are other spacing marks in use. In Hindi, Arabic and Pashto for example. (I don't have my reference examples around at the moment). So I think the unicode class `/\p{Zs}/ ` might be more appropriate here.

I believe `|[[:punct:]]` is a ruby specific implementation. To keep the implementation consistent between implementations I think it might be better to use Unicode character classes. All languages are able to understand these. For punctuation that would be `/\p{P}/`.

Reply:
For the Java implementation I went for `[\s\p{P}]` to determine if a character is whitespace or punctuation. I didn't look into Hindi/Arabic/Pashto spacing. I think we can defer supporting that until it becomes an issue. I prefer simplicity over completeness for this functionality - it's not execution, it's just snippet suggestions.

Reply:
Good enough for now I guess. I'll add the extra character classes + tests when I get the time. It doesn't add much complexity. It's all in the Unicode standard.

Reply:
Thanks for pointing that out, the Ruby implementation now uses \p{P} instead of [[:punct]]